### ISSUE TITLE
Color/indicate overlaps & combine "outlier" warning for terser/easier to read output

### ISSUE THREAD

**Issue by avar** [sharkdp/hyperfine#449]
Consider this benchmark which is designed to give us overlaps i.e. the first one sleeps for around `100..300ms`, the next for `200..400ms` etc:

```
$ hyperfine -L n 1,2,3,4 --shell /bin/bash -r 10 'sleep 0.$(({n} + $(($RANDOM % 3))))'
Benchmark 1: sleep 0.$((1 + $(($RANDOM % 3))))
  Time (mean ± σ):     210.7 ms ±  87.5 ms    [User: 0.7 ms, System: 0.0 ms]
  Range (min … max):   100.7 ms … 300.8 ms    10 runs
 
Benchmark 2: sleep 0.$((2 + $(($RANDOM % 3))))
  Time (mean ± σ):     310.7 ms ±  56.8 ms    [User: 0.5 ms, System: 0.4 ms]
  Range (min … max):   200.6 ms … 400.8 ms    10 runs
 
  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet PC without any interferences from other programs. It might help to use the '--warmup' or '--prepare' 
options.
 
Benchmark 3: sleep 0.$((3 + $(($RANDOM % 3))))
  Time (mean ± σ):     390.7 ms ±  73.8 ms    [User: 0.5 ms, System: 0.2 ms]
  Range (min … max):   300.7 ms … 500.8 ms    10 runs
 
Benchmark 4: sleep 0.$((4 + $(($RANDOM % 3))))
  Time (mean ± σ):     490.8 ms ±  99.5 ms    [User: 0.5 ms, System: 0.4 ms]
  Range (min … max):   400.7 ms … 600.9 ms    10 runs
 
Summary
  'sleep 0.$((1 + $(($RANDOM % 3))))' ran
    1.47 ± 0.67 times faster than 'sleep 0.$((2 + $(($RANDOM % 3))))'
    1.85 ± 0.85 times faster than 'sleep 0.$((3 + $(($RANDOM % 3))))'
    2.33 ± 1.08 times faster than 'sleep 0.$((4 + $(($RANDOM % 3))))'
```

The first time I saw the "statistical outliers" output I thought great, finally something that gives me information about results like this being rather useless:

```
1.27 ± 0.23
1.88 ± 0.39
```

I.e. `1.27+0.23 = 1.50`, and `1.88 - 0.39 = 1.49`, they overlap. So maybe it did get faster, or maybe it's just a fluke. The outlier detection on the other hand detects outliers *within* a given benchmark run. That's also useful. But what I'd really like is for the output to be like the above, but for the summary to instead end in say:

```
Summary
  'sleep 0.$((1 + $(($RANDOM % 3))))' ran
    #1: 1.47 ± 0.67 times faster than 'sleep 0.$((2 + $(($RANDOM % 3))))'
    #2; 1.85 ± 0.85 times faster than 'sleep 0.$((3 + $(($RANDOM % 3))))'
    #3: 2.33 ± 1.08 times faster than 'sleep 0.$((4 + $(($RANDOM % 3))))'
Overlaps:
 * #1 with #2 and #3
 * #2 with #1, #3 and #4
 * #3 with #2 and #3 
Outliers:
 * ...
```

In addition if there's overlaps the ± could be colored red instead of green, and maybe we should show "might be faster" instead of "[is] faster than".

**Comment by sharkdp** [comment]
Thank you for the feedback.

First off: I like the proposal!

But: I think your statistical interpretation is a bit off here. What hyperfine reports is the *mean* and the corrected sample standard deviation σ. A result like

* A: 100 ms ± 5 ms
* B: 101 ms ± 5 ms

*could* still very well mean that command A is faster (in a statistically significant way) than command B. A standard deviation of σ=5 ms means that there is a statistical fluctuation around the mean value of the order of 5 ms. That is: if you run the command *once*, there is a 68% chance that the runtime will be in the interval [95 ms, 105 ms].

However, if you want to know *how accurately we have measured the mean*, you are interested in the [*standard error*](https://en.wikipedia.org/wiki/Standard_error), which is σ/sqrt(N), where N is the number of benchmark runs. If we had performed N=100 runs, the standard error would be 5 ms/sqrt(100) = 0.5 ms, i.e. we would have measured the mean runtime with an accuracy of 0.5 ms. So it's quite likely that command A is indeed faster than command B, even though the 1σ intervals overlap.

My assumption is that you would be interested in this latter type of information. A rigorous way to do this is to perform [Welch's t-test](https://en.wikipedia.org/wiki/Welch%27s_t-test). We already have this sort of functionality, but it's currently hidden in hyperfines [Python scripts](https://github.com/sharkdp/hyperfine/tree/master/scripts). 

The script currently only works for two commands. Let's first start with a small number of benchmark runs:
``` bash
> hyperfine -L n 3,4 --shell /bin/bash -r 3 'sleep 0.$(({n} + $(($RANDOM % 3))))' --export-json overlap.json
> scripts/welch_ttest.py ~/overlap.json
Command 1: sleep 0.$((3 + $(($RANDOM % 3))))
Command 2: sleep 0.$((4 + $(($RANDOM % 3))))

t = 0.00968, p = 0.993

The two benchmarks are almost the same (p >= 0.05).
```
The script reports, that there is *no statistically significant difference* in the runtime of these two commands.

But things change if we perform more runs:
``` bash
> hyperfine -L n 3,4 --shell /bin/bash -r 50 'sleep 0.$(({n} + $(($RANDOM % 3))))' --export-json overlap.json
> scripts/welch_ttest.py ~/overlap.json
Command 1: sleep 0.$((3 + $(($RANDOM % 3))))
Command 2: sleep 0.$((4 + $(($RANDOM % 3))))

t = -7.62, p = 1.65e-11

There is a difference between the two benchmarks (p < 0.05).
```
Now we are extremely certain that the runtime of the two commands is different (which is indeed the case!).
